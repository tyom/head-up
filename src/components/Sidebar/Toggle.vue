<template>
  <button class="Toggle" @click="$emit('toggle')">
    <v-icon :name="toggleIconName" />
  </button>
</template>

<script>
export default {
  props: {
    toggled: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    toggleIconName() {
      return this.toggled ? 'chevron-left' : 'ellipsis-v';
    },
  },
};
</script>

<style scoped>
.Toggle {
  @apply absolute overflow-hidden z-10 inset-0 right-auto;
  @apply w-6 h-full text-gray-500 outline-none transition duration-100 cursor-default;
  box-shadow: inset -1px 0 #fff2;

  &:hover {
    @apply text-white bg-gray-800;
  }

  &:active {
    @apply bg-blue-900;
    transform: scaleY(1.2);
  }

  &.focus-visible {
    box-shadow: inset 0 0 0 1px #fff6;
  }
}
</style>
