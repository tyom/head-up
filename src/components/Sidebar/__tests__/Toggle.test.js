import { shallowMount } from '@vue/test-utils';
import Toggle from '../Toggle';

test('render untoggled', () => {
  const wrapper = shallowMount(Toggle, {
    propsData: {
      toggled: false,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('render toggled', () => {
  const wrapper = shallowMount(Toggle, {
    propsData: {
      toggled: true,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('responds to actions', () => {
  const wrapper = shallowMount(Toggle);

  wrapper.find('button').trigger('click');
  expect(wrapper.emitted('toggle')[0]).toBeTruthy();
});
