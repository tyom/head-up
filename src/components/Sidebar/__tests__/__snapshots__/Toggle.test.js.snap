// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render toggled 1`] = `
<button class="Toggle">
  <v-icon name="chevron-left"></v-icon>
</button>
`;

exports[`render untoggled 1`] = `
<button class="Toggle">
  <v-icon name="ellipsis-v"></v-icon>
</button>
`;
